`run_tests_jar` target does not work if `build` directory is not in `fuzion` repository directory

If I my current dir is `work` and the repo is in `../work` I get
```
 > PRECONDITIONS=true POSTCONDITIONS=true make -f ../fuzion/Makefile run_tests_jar

error 1: file does not exist: /home/fridi/fuzion/work/tests/hello/HelloWorld.fz

*** fatal errors encountered, stopping.
make: *** [../fuzion/Makefile:1121: run_tests_jar] Fehler 1

```
Ideally, we should have `jar` as a target for all or most tests to be able to fully test them, I was also thinking of adding an effects backend version for all tests. Ideas how to best do this are welcome (would prefer not to have many `skip_jar` or `skip_effects` files and also not unreasonably slow down the build process).

--- a/Makefile
+++ b/Makefile
@@ -1117,8 +1117,8 @@ run_tests_jvm_parallel: $(FZ_JVM) $(MOD_TERMINAL) $(MOD_LOCK_FREE) $(BUILD_DIR)/
 	$(FZ_SRC)/bin/run_tests_parallel.sh $(BUILD_DIR) jvm
 
 .PHONY .SILENT: run_tests_jar
-run_tests_jar: $(FZ_JVM)
-	$(BUILD_DIR)/bin/fz -jar tests/hello/HelloWorld.fz
+run_tests_jar: $(FZ_JVM) $(BUILD_DIR)/tests
+	$(BUILD_DIR)/bin/fz -jar $(BUILD_DIR)/tests/hello/HelloWorld.fz
 	java -jar HelloWorld.jar > /dev/null
 
 .PHONY: clean

